kura/org.eclipse.kura.rest.command.provider/src/main/java/org/eclipse/kura/internal/rest/command/CommandRestService.java

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.internal.rest.command;
+
+import static org.eclipse.kura.cloudconnection.request.RequestHandlerMessageConstants.ARGS_KEY;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.kura.KuraErrorCode;
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.cloud.app.command.CommandCloudApp;
+import org.eclipse.kura.cloud.app.command.KuraCommandRequestPayload;
+import org.eclipse.kura.cloud.app.command.KuraCommandResponsePayload;
+import org.eclipse.kura.cloudconnection.message.KuraMessage;
+import org.eclipse.kura.request.handler.jaxrs.DefaultExceptionHandler;
+import org.eclipse.kura.rest.command.api.RestCommandRequest;
+import org.eclipse.kura.rest.command.api.RestCommandResponse;
+import org.osgi.service.useradmin.Role;
+import org.osgi.service.useradmin.UserAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/command/v1")
+public class CommandRestService {
+
+    private static final Logger logger = LoggerFactory.getLogger(CommandRestService.class);
+
+    private static final String KURA_PERMISSION_REST_CONFIGURATION_ROLE = "kura.permission.rest.command";
+
+    private static final String PASSWORD_METRIC_NAME = "command.password";
+    public static final String RESOURCE_COMMAND = "command";
+
+    private CommandCloudApp commandCloudApp;
+
+    public void setUserAdmin(UserAdmin userAdmin) {
+        userAdmin.createRole(KURA_PERMISSION_REST_CONFIGURATION_ROLE, Role.GROUP);
```

Comment:
What if the Configuration Role already exists ?

Reply:
![image](https://github.com/eclipse/kura/assets/29900100/c006f533-e2f7-4f82-8a93-0d716a230fc1)
From OSGi 7 spec, I think this should be okay?